File: stian/day_one/go/day_one.go
package main

import (
	"fmt"
	"os"
)

func main() {
	file, err := os.Open("day_one.txt")
	if err != nil {
		panic(err)
	}

	defer file.Close()
	var line int
	data := make([]int, 0)
	_, err = fmt.Fscanf(file, "%d", &line)

	for err == nil {
		data = append(data, line)
		_, err = fmt.Fscanf(file, "%d\n", &line)
	}

	fmt.Println("Part 1:", partOne(data))
	fmt.Println("Part 2:", partTwo(data))
}

func partOne(data []int) int {
	prev := data[0]
	counter := 0
	for _, v := range data {
		if v > prev {
			counter++
		}
		prev = v
	}
	return counter
}

func partTwo(data []int) int {
	curr_sum := 0
	prev_sum := 0
	counter := 0
	for i, v := range data {
		curr_sum += v
		if i > 2 {
			if curr_sum > prev_sum {
				counter++
			}
			prev_sum = curr_sum
			curr_sum -= data[i-2]
		}
	}
	return counter
}

File: stian/day_one/java/dayOne.java
package stian.day_one.java;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.Scanner;

public class dayOne {
    public static void main(String[] args) throws FileNotFoundException {
        Scanner s = new Scanner(new File("day_one.txt"));
        ArrayList<String> list = new ArrayList<String>();
        while (s.hasNext()){
            list.add(s.next());
        }
        s.close();
        System.out.println("Part One: " + partOne(list));
        System.out.println("Part TWo: " + partTwo(list));
    }

    public static int partOne(ArrayList<String> data) {
        int prev = Integer.parseInt(data.get(0));
        int counter = 0;
        for (int i = 0; i < data.size(); i++) {
            int curr = Integer.parseInt(data.get(i));
            if (curr > prev) {
                counter++;
            }
        }
        return counter;
    }

    public static int partTwo(ArrayList<String> data) {
        int curr = 0;
        int prev = 0;
        int counter = 0;
        for (int i = 0; i < data.size(); i++) {
            curr += Integer.parseInt(data.get(i));
            if (i > 2) {
                if (curr > prev) {
                    counter++;
                }
                prev = curr;
                curr -= Integer.parseInt(data.get(i-2));
            }
        }
        return counter;
    }
}
